test/byContract/IexecEscrow/IexecEscrowNative.ts

```diff
@@ -0,0 +1,280 @@
+            iexecPocoAsAccountA = iexecPoco.connect(accountA);
+            iexecPocoAsAdmin = iexecPoco.connect(iexecAdmin);
+        }
+
+        describe('Receive and Fallback', () => {
+            it('Should call receive successfully', async () => {
+                await expect(
+                    accountA.sendTransaction({
+                        to: iexecPoco.address,
+                        value: depositNativeAmount,
+                    }),
+                )
+                    .to.changeEtherBalances(
+                        [accountA, iexecPoco],
+                        [-depositNativeAmount, depositNativeAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [accountA], [depositAmount])
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(AddressZero, accountA.address, depositAmount);
+            });
+
+            it('Should call fallback successfully', async () => {
+                const randomData = ethers.utils.hexlify(
+                    ethers.utils.toUtf8Bytes((Math.random() * 0xfffff).toString(16)),
+                );
+
+                await expect(
+                    accountA.sendTransaction({
+                        to: iexecPoco.address,
+                        value: depositNativeAmount,
+                        data: randomData,
+                    }),
+                )
+                    .to.changeEtherBalances(
+                        [accountA, iexecPoco],
+                        [-depositNativeAmount, depositNativeAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [accountA], [depositAmount])
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(AddressZero, accountA.address, depositAmount);
+            });
+        });
+
+        describe('Deposits', () => {
+            it('Should deposit native tokens', async () => {
+                expect(await iexecPocoAsAccountA.callStatic.deposit(...depositArgs)).to.be.true;
+                await expect(iexecPocoAsAccountA.deposit(...depositArgs))
+                    .to.changeEtherBalances(
+                        [accountA, iexecPoco],
+                        [-depositNativeAmount, depositNativeAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [accountA], [depositAmount])
+                    .to.emit(iexecPoco, 'Transfer')
+                    .withArgs(AddressZero, accountA.address, depositAmount);
+            });
+
+            it('Should deposit native tokens and return excess', async () => {
+                const depositAmountExecess = 0.1;
+                const depositAmountWithExecess = 100.1;
+                const expectedTokenTransfer = depositAmountWithExecess - depositAmountExecess;
+
+                const depositNativeAmountExecess = ethers.utils.parseUnits('0.1', 9);
+                const depositNativeAmountWithExecess = ethers.utils.parseUnits('100.1', 9);
```

Comment:
Use previous vars instead of '0.1'

Reply:
Actually the excess part is when we call it with float amount so we need to introduce 'x.y'

Reply:
`parseUnits(depositAmountExecess`

Reply:
Same com than Zied https://github.com/iExecBlockchainComputing/PoCo/pull/99/files#r1703713549

Reply:
Done [here](https://github.com/iExecBlockchainComputing/PoCo/pull/99/files#diff-8da3ad42fa3f9c43ed191339ed8cd186995c3902ec6cf125342c2c71779f5fefR96) tell me if that's ok for you